crm-app/src/component/ui/ProfileView.js

```diff
@@ -25,78 +26,72 @@ class ProfileView extends Component {
 	constructor (props) {
 		super(props);
 		this.formRef = React.createRef();
-		this.isFilledForm = false;
+		this.isWritten = false;
 	}
 
 	handleSave = (e) => {
-		const name = e.target.form.name.value;
-		const mail = e.target.form.mail.value;
-		const phone = e.target.form.phone.value;
-
-		if (name !== '' || mail !== '' || phone !== '') {
-			const info = {
-				name,
-				mail,
-				phone
-			}
-			const retval = this.props.onSave(mail, info, this.isFilledForm);
-			if (!retval) {
-				this.clearForm(e.target);
-			}
+		const { displayedIndex, onAdd, onUpdate } = this.props;
+		const form = e.target.form;
+		const name = form.name.value;
+		const mail = form.mail.value;
+		const phone = form.phone.value;
+
+		if (name === '' && mail === '' && phone === '')
+			return;
+
+		// check if it is new customer or modified customer
+		if (this.isWritten) {
+			onUpdate(displayedIndex, name, phone);
+		} else {
+			if (displayedIndex < 0) {
+				onAdd(name, mail, phone);
+			} else
+				alert("Error: This email alreay exists.");
```

Comment:
Please wrap the code with `{}`.

Reply:
A single quotation would be better here.

Reply:
Fixed